consumer/consumer.go

```diff
@@ -30,17 +30,28 @@ type MetricsConsumer interface {
 	ConsumeMetricsData(ctx context.Context, md consumerdata.MetricsData) error
 }
 
+// MetricsConsumerV2 is the new metrics consumer interface that receives data.MetricData, processes it
+// as needed, and sends it to the next processing node if any or to the destination.
+type MetricsConsumerV2 interface {
+	ConsumeMetricsData(ctx context.Context, md data.MetricData) error
+}
+
+// BaseTraceConsumer defines a common interface for TraceConsumer and TraceConsumerV2
```

Comment:
```suggestion
// BaseTraceConsumer defines a common interface for TraceConsumer and TraceConsumerV2.
```

Reply:
We try to have all comments (especially with of public elements) with proper punctuation so the text reads well on godoc.